controllers/reservation.controllers.js

```diff
@@ -559,8 +555,91 @@ const submitReserve = async ({
   });
 };
 
+const deleteReservation = async ({
+  client, ack, say, body,
+}) => {
+  await ack();
+  const { text } = body;
+
+  try {
+    if (text) {
+      const {
+        date, roomName, user, currentDate, formattedDate,
+      } = await basicInformation(text, body);
+
+      if (!user) {
```

Comment:
some of these validations are the exact same as in addReservation, would it make sense to include them as part of basicInformation()?